src/schemes.hpp

```diff
@@ -39,6 +39,7 @@ class CoreMPL {
 public:
     CoreMPL() = delete;
     CoreMPL(const std::string& strId) : strCiphersuiteId(strId) {}
+    virtual ~CoreMPL() {}
```

Comment:
is this still necessary when you make the other types final?

Reply:
Yes, because final in other class mean just no more inheritance and it doesn't need it's own virtual destructor. But above class/struct still needs so

Reply:
ok. I guess we already have other virtual functions in this class, and we *do* use it polymorphically. so this seems right.